app/com/linkedin/drelephant/tony/heuristics/TaskGPUHeuristic.java

```diff
@@ -0,0 +1,115 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.drelephant.tony.heuristics;
+
+import com.linkedin.drelephant.analysis.Heuristic;
+import com.linkedin.drelephant.analysis.HeuristicResult;
+import com.linkedin.drelephant.analysis.HeuristicResultDetails;
+import com.linkedin.drelephant.analysis.Severity;
+import com.linkedin.drelephant.configurations.heuristic.HeuristicConfigurationData;
+import com.linkedin.drelephant.tony.data.TonyApplicationData;
+import com.linkedin.drelephant.tony.data.TonyTaskData;
+import com.linkedin.drelephant.tony.util.TonyUtils;
+import com.linkedin.drelephant.util.Utils;
+import com.linkedin.tony.Constants;
+import com.linkedin.tony.TonyConfigurationKeys;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.log4j.Logger;
+
+
+/**
+ * For each type of task (e.g.: worker, ps), this heuristic checks the GPU utilization across all tasks of that type
+ * and compares against some thresholds. The final severity is the highest severity across all task types.
+ */
+public class TaskGPUHeuristic implements Heuristic<TonyApplicationData> {
+  private static final Logger _LOGGER = Logger.getLogger(TaskGPUHeuristic.class);
+  private static final String[] MAX_METRICS_TO_APPLY = {
+      Constants.MAX_GPU_UTILIZATION,
+      Constants.MAX_GPU_FB_MEMORY_USAGE,
+      Constants.MAX_GPU_MAIN_MEMORY_USAGE};
+  private static final String[] AVG_METRICS_TO_APPLY = {
+      Constants.AVG_GPU_UTILIZATION,
+      Constants.AVG_GPU_FB_MEMORY_USAGE,
+      Constants.AVG_GPU_MAIN_MEMORY_USAGE
+  };
+
+  private HeuristicConfigurationData _heuristicConfData;
+
+  /**
+   * Constructor for {@link TaskGPUHeuristic}.
+   * @param heuristicConfData  the configuration for this heuristic
+   */
+  public TaskGPUHeuristic(HeuristicConfigurationData heuristicConfData) {
+    this._heuristicConfData = heuristicConfData;
+  }
+
+  @Override
+  public HeuristicResult apply(TonyApplicationData data) {
+    _LOGGER.debug("Applying TaskGPUHeuristic");
+    Map<String, Map<Integer, TonyTaskData>> taskMap = data.getTaskMap();
+    Configuration conf = data.getConfiguration();
+
+    Set<String> taskTypes = com.linkedin.tony.util.Utils.getAllJobTypes(conf);
+    Severity finalSeverity = Severity.NONE;
+    List<HeuristicResultDetails> details = new ArrayList<>();
+
+    for (String taskType : taskTypes) {
+      details.add(new HeuristicResultDetails("Number of " + taskType + " tasks",
```

Comment:
Can you also print the number of GPUs requested per task type? If it's 0, I think you shouldn't show any of the metric information.